src/TableRow.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from "react";
+import moment from 'moment';
+
+class TableRow extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isRowSelected: false
+        }
+    }
+
+    handleClick = (event) => {
+        const clickedRow = event.target.parentElement;
+        const previousState = this.state.isRowSelected; clickedRow.className = previousState ? '' : 'selected';
```

Comment:
Just realised I posted a comment on some outdated code. Please ignore that comment.

This is something that you have to be careful about in React: you're directly manipulating the DOM. This is potentially a problem because now React doesn't know that your change is deliberate and will wipe away your change next time your render. Your manual DOM change is in the "real DOM", but not in React's virtual DOM and so it removes your manual change.

You've got 99% of the way there by splitting out a `TableRow` component. If you instead use the `selected` state in your `render()` method to add/remove this class then you'll be good! Well done figuring this problem out - it's intentionally quite tricky!

Reply:
Thanks for your time and feedback Ali, I will implement your feedback.